pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
Nope. Still doesn't make sense. If you want them to have a `Validate` method, but optionally have these setters, then have `Initialize` take something that requires a `Validate` method. If you want them to validate the setting of these things, then have the setters return an error.

Where is `Validate` called anyway? Openshift only? If so, can I have a link?